pkg/smokescreen/smokescreen.go

```diff
@@ -497,39 +516,34 @@ func BuildProxy(config *Config) *goproxy.ProxyHttpServer {
 func logProxy(config *Config, pctx *goproxy.ProxyCtx) {
 	sctx := pctx.UserData.(*smokescreenContext)
 
-	var contentLength int64
-	if pctx.Resp != nil {
-		contentLength = pctx.Resp.ContentLength
-	}
-
-	fromHost, fromPort, _ := net.SplitHostPort(pctx.Req.RemoteAddr)
-
-	fields := logrus.Fields{
-		"src_host":       fromHost,
-		"src_port":       fromPort,
-		"content_length": contentLength,
-	}
-
-	if sctx.decision.resolvedAddr != nil {
-		fields["dst_ip"] = sctx.decision.resolvedAddr.IP.String()
-		fields["dst_port"] = sctx.decision.resolvedAddr.Port
-	}
+	fields := logrus.Fields{}
 
 	// attempt to retrieve information about the host originating the proxy request
-	fields["src_host_common_name"] = "unknown"
-	fields["src_host_organization_unit"] = "unknown"
 	if pctx.Req.TLS != nil && len(pctx.Req.TLS.PeerCertificates) > 0 {
-		fields["src_host_common_name"] = pctx.Req.TLS.PeerCertificates[0].Subject.CommonName
+		fields["inbound_remote_x509_cn"] = pctx.Req.TLS.PeerCertificates[0].Subject.CommonName
 		var ouEntries = pctx.Req.TLS.PeerCertificates[0].Subject.OrganizationalUnit
 		if len(ouEntries) > 0 {
-			fields["src_host_organization_unit"] = ouEntries[0]
+			fields["inbound_remote_x509_ou"] = ouEntries[0]
 		}
 	}
 
 	decision := sctx.decision
 	if sctx.decision != nil {
 		fields["role"] = decision.role
 		fields["project"] = decision.project
+	}
+
+	// add the above fields to all future log messages sent using this smokescreen context's logger
+	sctx.logger = sctx.logger.WithFields(fields)
+
+	// start a new set of fields used only in this log message
+	fields = logrus.Fields{}
+
+	if pctx.Resp != nil {
+		fields["content_length"] = pctx.Resp.ContentLength
+	}
+
+	if sctx.decision != nil {
```

Comment:
Could we roll this up into the block above which checks for `sctx.decision != nil`?

Reply:
If we do that, then those three fields (`decision_reason`, `enforce_would_deny`, and `allow`) would get added to all future log messages written by this smokescreen context's `logger` instead of just this `CANONICAL-PROXY-DECISION` message.

Reply:
Yeah, sorry about that. My eyes skipped over that being reassigned to `sctx.logger`.